src/main/java/org/openstreetmap/atlas/geography/atlas/sqlite/PackedAtlasFieldDelegator.java

```diff
@@ -0,0 +1,152 @@
+package org.openstreetmap.atlas.geography.atlas.sqlite;
+
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedTagStore;
+import org.openstreetmap.atlas.utilities.compression.IntegerDictionary;
+import org.openstreetmap.atlas.utilities.maps.LongToLongMap;
+
+/**
+ * This delegator allows the {@link SQLiteWriter} to access internal {@link PackedAtlas} fields it
+ * needs to efficiently build the database.
+ *
+ * @author lcram
+ */
+public class PackedAtlasFieldDelegator
+{
+    // TODO this class has a lot of fields. It may be worth refactoring it out and just using
+    // reflection. Alternatively, we could also just make a ton of PackedAtlas fields public. Let's
+    // discuss these options please.
+
```

Comment:
I think the reason the getters are not public in PackedAtlas is that in theory, you could mutate what gets returned. But, we're not... However, making a copy is also a bad idea, because we don't want to abuse our memory usage.

Personally, I'd just make the getters public in packed atlas, but I'm fine with whatever, as long as we don't do a copy.

Reply:
Yeah, no copies for sure. If this were C/C++, we would have `const` here to help us out, and all parties would be satisfied...

I'm not particularly committed to any of the particular solutions, but I do sympathize with @matthieun 's desire to keep `PackedAtlas` as locked down as possible.